frontend/integration-tests/protractor.conf.ts

```diff
@@ -228,7 +228,8 @@ export const checkErrors = async () =>
     }
   });
 
-export const firstElementByTestID = (id: string) => $(`[data-test-id=${id}]`);
+export const firstElementByTestID = (id: string) =>
+  element.all(by.css(`[data-test-id=${id}]`)).first();
```

Comment:
Could you explain why this needed to change?

Reply:
Yes, the integration tests was throwing a 'multiple instances detected but first one will be used' when I was using just the `$`

Reply:
👍

Reply:
I'm not sure we should be suppressing that warning. In a lot of cases, it's not expected, and you'd want to know. I'd suggest separate helpers when we expect there to be more than one. Something like (untested)

```ts
export const elementByTestID = (id: string) => $(`[data-test-id=${id}]`);
export const firstElementByTestID = (id: string) => $$(`[data-test-id=${id}]`)[0];
```

Reply:
I'd even suggest

```ts
export const allElementsByTestID = (id: string) => $$(`[data-test-id=${id}]`);
```

Then call it like `allElementsByTestID('my-id')[0]`. This makes it very clear what's happening.

Is there a way we can make the test ID unique on the page, though, to avoid this problem?

Reply:
I like `firstElementByTestID = (id: string) => $$(`[data-test-id=${id}]`)[0];`

Reply:
I'm using `firstElementByTestID` to get the first drop-down menu item, probably not the best if menu order changes.  I'll look into being able to pass in `data-test-id` to the dropdown component.

Reply:
> I'm using `firstElementByTestID` to get the first drop-down menu item, probably not the best if menu order changes. I'll look into being able to pass in `data-test-id` to the dropdown component.

+1, this is fragile. I would add a specific `data-test-id`

Reply:
Ok, was able to replace it with:
`await firstElementByTestID('dropdown-menu-pagerduty_configs').click();` 
after updating [this](https://github.com/openshift/console/pull/3698/commits/08ec3c02d2937270338ffd5685a68923621f4f99#diff-447b013466c334bed9fb476dc38fc194R126) and [this](https://github.com/openshift/console/pull/3698/commits/08ec3c02d2937270338ffd5685a68923621f4f99#diff-73ded5241d763da982ede2d7c59d3c94R10).  Also reverted changes to `firstElementByTestID`.